R/integrations.R

```diff
@@ -80,46 +80,80 @@ seuratProcV3 <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, n.pcs
 #' @param output.path path to a folder, where intermediate files will be saved
 #' @param metadata.df data.frame with additional metadata with rownames corresponding to cell ids, which should be passed to ScanPy.
 #' If NULL, only information about cell ids and origin dataset will be saved.
-#' @param n.dims to emulate PCA, Conos embed the graph to a space with `n.dims` dimensions and saves it as PCA
-#' @param verbose print more messages
+#' @param norm logical, include the matrix of normalised counts. Default: FALSE
+#' @param embed logical, include the current conos embedding. Default: FALSE
+#' @param pseudo.pca logical, produce an emulated PCA by embedding the graph to a space with `n.dims` dimensions and save it as a pseudoPCA. Default: FALSE
+#' @param pca logical, include PCA of all the samples (not batch corrected). Default: FALSE
+#' @param n.dims number of dimensions for calculating PCA and/or pseudoPCA
+#' @param connect logical, include graph of connectivities and distances. Default: FALSE
+#' @param verbose verbose mode. Default: FALSE
 #'
 #' @export
-saveConosForScanPy <- function(con, output.path, metadata.df=NULL, n.dims=100, verbose=F) {
+saveConosForScanPy <- function(con, output.path, metadata.df=NULL, norm=FALSE, pseudo.pca=FALSE, pca=FALSE, n.dims=100, embed=FALSE, connect=FALSE, verbose=FALSE) {
```

Comment:
My point here was that I wanted the function to be the most compatible by default. Someone may run it without creating an embedding, at which point it wouldn't work. Saving a normalised matrix takes the longest time (because of float numbers), so I'm not sure it should be a default behaviour as well.   

I can see you point out the problem with missing embedding in your other comment. I can add a quick check for `NULL` there. 

My opinion is that by default it should only give a raw matrix with names of cells and genes - ready to be analysed in a different tool. Other options can be turned on depending on user preference. At the end of the day, if they copy the command from the tutorial, they will have everything set to `TRUE`.

However, you are the major contributor to the package. I'll change the default options if you say so.

Reply:
Agree with the logic, that's fair. Though I think that the main entity of Conos is the graph, not the matrix. And embedding is also important and small. Let's set `embed` and `connect` to `TRUE` and if it's not estimated, show a warning and ignore it.
Btw, can you please rename `embed`, `norm` and `connect` to `embedding`, `cm.norm` and `alignment.graph`?

Reply:
What do you think would be the best way to check for whether user created the embedding or not? I just noticed that simple evaluating `if (!is.null(con$embedding))` doesn't work (as I thought), since `$embedding` is an S4 object even just right after calling `con$new()`. Would checking for length always work? Something like `if (length(con$embedding)>1)`

Reply:
`(length(con$embedding)>1)` seems to be working. 

I accommodated the proposed changes in 00d3633, 282fb0d and 079f143.

I think the PR can be merged now.

Reply:
Yes, that should work